backend/tests/alert/test_alert.py

```diff
@@ -69,37 +71,281 @@ def override_delay(modules_names, before_func, before_kwargs):
+        self.assertFalse(mock_push_notification.called)
+        r_legacy = Registration.objects.get(id=self.r_legacy.id)
+        self.assertTrue(r_legacy.notification_sent)
+        self.assertEqual("ADM", r_legacy.notification_sent_by)
+
+    def send_alert_helper(self, mock_email, mock_text, mock_push_notification, push_notification):
+        self.r.user.profile.push_notifications = push_notification
+        self.r.user.profile.save()
+        r = Registration.objects.get(id=self.r.id)
+        self.assertIsNotNone(r.user)
+        self.assertIsNotNone(r.user.profile)
+        self.assertEquals("yo@example.com", r.user.profile.email)
+        self.assertEquals("+15555555555", r.user.profile.phone)
+        self.assertEquals("CIS-160-001", r.section.full_code)
+        self.assertIsNone(r.email)
+        self.assertIsNone(r.phone)
+        self.assertEquals(push_notification, r.user.profile.push_notifications)
+        self.assertFalse(r.notification_sent)
+        self.assertTrue(r in get_active_registrations("CIS-160-001", TEST_SEMESTER))
+        self.assertTrue(
+            r in get_registrations_for_alerts("CIS-160-001", TEST_SEMESTER, course_status="O")
+        )
+        self.assertFalse(
+            r in get_registrations_for_alerts("CIS-160-001", TEST_SEMESTER, course_status="C")
+        )
+        self.assertEquals(
+            0, len(get_registrations_for_alerts("CIS-160-001", TEST_SEMESTER, course_status="X"))
+        )
+        self.assertEquals(
+            0, len(get_registrations_for_alerts("CIS-160-001", TEST_SEMESTER, course_status=""))
+        )
+        tasks.send_alert(self.r.id, sent_by="ADM")
+        r = Registration.objects.get(id=self.r.id)
+        self.assertTrue(mock_email.called)
+        self.assertEquals(not push_notification, mock_text.called)
+        self.assertEquals(push_notification, mock_push_notification.called)
+        self.assertTrue(r.notification_sent)
+        self.assertIsNotNone("ADM", r.notification_sent_by)
+        self.assertEqual("ADM", r.notification_sent_by)
+        self.assertFalse(r in get_active_registrations("CIS-160-001", TEST_SEMESTER))
+        self.assertFalse(
+            r in get_registrations_for_alerts("CIS-160-001", TEST_SEMESTER, course_status="O")
+        )
+        self.assertFalse(
+            r in get_registrations_for_alerts("CIS-160-001", TEST_SEMESTER, course_status="C")
+        )
 
-    def test_dont_resend_alert(self, mock_email, mock_text):
+    def test_send_alert_push(self, mock_email, mock_text, mock_push_notification):
+        self.send_alert_helper(
+            mock_email, mock_text, mock_push_notification, push_notification=True
+        )
+
+    def test_send_alert(self, mock_email, mock_text, mock_push_notification):
+        self.send_alert_helper(
+            mock_email, mock_text, mock_push_notification, push_notification=False
+        )
+
+    def send_close_notification_helper(
```

Comment:
Again, docs on what the helper is doing would be great. Seems to be doing a lot that's hard to parse at first glance!